feature_engine/selection/shuffle_features.py

```diff
@@ -0,0 +1,213 @@
+            raise ValueError("cv can only take True or False")
+
+        if not isinstance(regression, bool):
+            raise ValueError("regression can only take True or False")
+ 
+        self.features_to_shuffle = \
+            _define_variables(features_to_shuffle)
+    
+        self.estimator = estimator
+        
+        self.scoring = scoring
+        
+        self.threshold = threshold
+        
+        self.regression = regression
+        
+        self.cv = cv
+
+    def fit(self, X, y):
+        """
+
+        Parameters
+        ----------
+        
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The input dataframe
+
+        y: array-like of shape (n_samples) 
+            Target variable. Required to train the estimator.
+
+
+        Attributes
+        ----------
+
+        shuffled_features_: dict
+            The shuffled features values
+
+        """
+
+        # check input dataframe
+
+        X = _is_dataframe(X)
+        
+        # find all variables or check those entered are in the dataframe
+        self.features_to_shuffle = _find_all_variables(X, self.features_to_shuffle)
+        
+        # Fit machine learning model with the input estimator if provided.
+        # If the estimator is not provided, default to random tree model 
+        # depending on value of self.regression
+        
+        if (self.estimator is not None):
+            
+            model = self.estimator.fit(X,
+                                       y)
+            y_pred = 1
+            model_performance = 1
+            
+        elif (self.regression):
+            
+            model = RandomForestRegressor().fit(X,
+                                                y)
+            y_pred = 1
+            model_performance = 1
+            
+        else:
+            
+            model = RandomForestClassifier().fit(X,
+                                                 y)
+            y_pred = model.predict_proba(X)[:, 1]
+            model_performance = roc_auc_score(y,
+                                              y_pred)
+
+
+        
+        # dict to collect features and their performance_drift 
+        
+        self.features_performance_drifts_ = {}
+
+        # shuffle features and save feature performance drift into a dict
+
+        for feature in self.features_to_shuffle:
+
+        #  Create a copy of X
+            X_shuffled = X.copy()
+                                       
+            # shuffle individual feature
+            X_shuffled[feature] = X_shuffled[feature].sample(frac=1).reset_index(drop=True)
+            
+            # Use fitted model to calculate new target prediction with
+            # the shuffled feature.
+            # Then, calculate this model performance.
+            if (self.regression):
+                
+                shuffled_y_pred = model.predict(X_shuffled)
+                shuffled_model_performance = 1
+                
```

Comment:
here we would replace this bit by the fit estimators from cross_validate. Is this possible?

Reply:
I'm going to work on this next.

They way things are coded now is as follows: ( I'm going to push my code in a few)
fit an estimator for original dataset using cross validation
calculate model performance as the average of each folds' error.

for each feature:
fit the estimator again with the dataset having the feature  features
calculate new model performance as the average of each folds' error.
calculate model drift for each feature.

@solegalli